File: csharp/factory_method/test/stores/MilkStoreTest.cs
using factory_method.main;
using factory_method.main.stores;
using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.Text;

namespace factory_method.test.stores
{
    [TestFixture]
    class MilkStoreTest
    {
        MilkStore milkStore;
        Cone cone;

        [SetUp]
        public void SetUp()
        {
            milkStore = new MilkStore();
            cone = new Cone();
        }
        [TestCase]
        public void MilkTest()
        {
            cone = milkStore.prepareCone("Guava");
            Assert.AreEqual("Guava Milk Ice Cream", cone.getIceCream().getName());
            cone = milkStore.prepareCone("Mango");
            Assert.AreEqual("Mango Milk Ice Cream", cone.getIceCream().getName());
            cone = milkStore.prepareCone("Pineapple");
            Assert.AreEqual("Pineapple Milk Ice Cream", cone.getIceCream().getName());
            cone = milkStore.prepareCone("Strawberry");
            Assert.AreEqual("Strawberry Milk Ice Cream", cone.getIceCream().getName());
        }

    }
}

File: java/src/test/java/mx/iteso/MilkStoreTest.java
package mx.iteso;

import org.junit.Test;
import static org.junit.Assert.assertEquals;
import mx.iteso.Cone;
import mx.iteso.stores.MilkStore;

public class MilkStoreTest {

    @Test
    public void testGuavaMilkIceCream() {
        Cone cone = new MilkStore().prepareCone("Guava");
        assertEquals(cone.getIceCream().getName(), "Guava Milk Icecream");
    }
    @Test
    public void testMangoMilkIceCream() {
        Cone cone = new MilkStore().prepareCone("Mango");
        assertEquals(cone.getIceCream().getName(), "Mango Milk Icecream");
    }
    @Test
    public void testPineappleMilkIceCream() {
        Cone cone = new MilkStore().prepareCone("Pineapple");
        assertEquals(cone.getIceCream().getName(), "Pineapple Milk Icecream");
    }
    @Test
    public void testStrawberryMilkIceCream() {
        Cone cone = new MilkStore().prepareCone("Strawberry");
        assertEquals(cone.getIceCream().getName(), "Strawberry Milk Icecream");
    }

}